### PR TITLE
[debian-ha] Release citus-ha 2.0.1

### PR DESCRIPTION
**Pull request by thanodnl**
This patch, together with https://github.com/citusdata/citus-ha/pull/186 allows debian packages to be created both for pg10 and pg11 at the same time.

This can be tested with:
```bash
$ citus_package -p=ubuntu/bionic local cli_destdir
```

### CODE REVIEW

File: debian/rules [hunk 1]
```diff
@@ -3,26 +3,19 @@
 include /usr/share/postgresql-common/pgxs_debian_control.mk
 
 override_dh_auto_build:
-	+make PG_CONFIG=/usr/lib/postgresql/11/bin/pg_config
-	+make
-	+cd src/monitor && pg_buildext build build-%v
+	+pg_buildext build $(shell pwd)/src/bin/citusha build-%v
```
Comment (DimCitus) [inline debian/rules]: What about the alternative packaging which consists of having a separate `citusha` and `postgresql-X-citusha` packages from a single `debian/control.in` file. In debian you can have multiple binary packages built from the same source package. I think that would be the best way to deal with the problem.

**Review by DimCitus — CHANGES_REQUESTED** [review]

File: debian/rules (hunk 1 above)
Comment (thanodnl) [reply debian/rules]: done,

given that the extension was already called `citus-ha` I went for the same in the new package.

File: debian/rules [hunk 2]
```diff
@@ -1,28 +1,26 @@
...
-	+dh_install --sourcedir=$(shell pg_config --bindir)
-
 override_dh_auto_install:
-	+make -C src/bin install
-	+cp ./src/monitor/citusha* ./src/monitor/build-*/
-	+cd src/monitor && pg_buildext install build-%v postgresql-%v-citus-ha
-	+cp -R ./src/monitor/* ./
+	+make -C $(shell pwd)/src/bin/citusha install \
+		DESTDIR=$(CURDIR)/debian/citus-ha \
+		BINDIR=/usr/bin
```
Comment (DimCitus) [inline debian/rules:22]: Do we need both DESTDIR and BINDIR here?

**Review by DimCitus — APPROVED** [review]

File: debian/rules (hunk 2 above)
Comment (thanodnl) [reply debian/rules:22]: Here not necessarily as they are concatenated in the Makefile, however they serve different purposes so I think it is useful to have them both in the Makefile.

By using them here I think it is easier to understand the binary will end up in `/usr/bin`. `DESTDIR` is to put it in the directory that will be packaged for the `deb`

**Comment by thanodnl** [comment]
Once we merge this it will rebuild the packages.
So I think we need to do a point release for at least the debian family.

Any problems if I release a 2.0.1 version with only the changes to the Makefile from the 2.0.0 version?

**Comment by DimCitus** [comment]
> Any problems if I release a 2.0.1 version with only the changes to the Makefile from the 2.0.0 version?

That's what point releases are all about, so no problem that I can foresee no. We can even make more of them if needed and then we'll say we applied lots of QA to this packaging... which is what I feel like we're doing, and that's good!

**Comment by thanodnl** [comment]
Ready for squash commit after release is tagged ( requires rerun of travis to find the tag to release, and verify everything builds )

File: debian/changelog [hunk 3]
```diff
@@ -1,3 +1,12 @@
+citus-ha (2.0.1) testing; urgency=low
+
+  * Build system improvements
+
+  * Implement support for VPATH builds, BINDIR, and DESTDIR
+
+ -- Nils Dijk <nils@citusdata.com>  Fri, 22 Feb 2019 17:15:00 +0100
```
Comment (thanodnl) [inline debian/changelog]: wrong date

File: debian/control.in [hunk 4]
```diff
@@ -7,6 +7,16 @@ Build-Depends: debhelper (>= 9), autotools-dev, flex, postgresql-server-dev-all
...
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}
+Provides: citus-ha
+Conflicts: citus-ha
+Recommends: postgresql-10-citus-ha | postgresql-11-citus-ha
+Suggests: postgresql-11-citus-8.1 | postgresql-10-citus-8.1 | postgresql-10-citus-enterprise-8.1 | postgresql-11-citus-enterprise-8.1
+Description: Citus high availability CLI
+ This CLI is used to manage HA installations of a Citus cluster.
+
 Package: postgresql-PGVERSION-citus-ha
```
Comment (thanodnl) [inline debian/control.in:20]: add requirement to citus-ha

File: debian/rules [hunk 5]
```diff
@@ -1,28 +1,26 @@
...
+	+pg_buildext install $(shell pwd)/src/monitor build-%v postgresql-%v-citus-ha
 
 %:
-	dh $@
+	dh $@
```
Comment (thanodnl) [inline debian/rules]: new line

File: debian/control.in (hunk 4 above)
Comment (thanodnl) [reply debian/control.in:20]: added `citus-ha` to `Depends:`, verified in docker that it depends on it and is installable together with that package

**Pull request merged**